extensions/librdkafka/PublishKafka.cpp

```diff
@@ -141,6 +141,59 @@ void PublishKafka::initialize() {
 
 void PublishKafka::onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory> &sessionFactory) {
   interrupted_ = false;
+
+  // Try to get a KafkaConnection
+  std::string client_id, brokers;
+  if (!context->getProperty(ClientName.getName(), client_id)) {
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Client Name property missing or invalid");
+  }
+  if (!context->getProperty(SeedBrokers.getName(), brokers)) {
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Known Brokers property missing or invalid");
+  }
+
+  // Get some properties not (only) used directly to set up librdkafka
+  std::string value;
+
+  // Batch Size
+  value = "";
+  if (context->getProperty(BatchSize.getName(), batch_size_)) {
+    logger_->log_debug("PublishKafka: Batch Size [%lu]", batch_size_);
+  } else {
+    batch_size_ = 10;
+  }
+
+  // Target Batch Payload Size
+  value = "";
+  if (context->getProperty(TargetBatchPayloadSize.getName(), target_batch_payload_size_)) {
+    logger_->log_debug("PublishKafka: Target Batch Payload Size [%llu]", target_batch_payload_size_);
+  } else {
+    target_batch_payload_size_ = 512 * 1024U;
+  }
+
+  // Max Flow Segment Size
+  value = "";
+  if (context->getProperty(MaxFlowSegSize.getName(), max_flow_seg_size_)) {
+    logger_->log_debug("PublishKafka: Max Flow Segment Size [%llu]", max_flow_seg_size_);
+  } else {
+    max_flow_seg_size_ = 0U;
```

Comment:
Instead of doing this, why don't we simply add default values to these properties, so getProperty always succeeds?

Reply:
Fixed, we already had default value, maybe we were overcareful